client/material/button.tsx

```diff
@@ -171,47 +197,73 @@ export function useButtonState({
   )
 
   const keyDownActivatedRef = useRef(false)
-  const handleKeyDown = useCallback(
-    (event: React.KeyboardEvent) => {
-      const activationTarget = event.currentTarget
-      if (activationTarget.matches(':active')) {
-        rippleRef.current?.onActivate(event)
-        keyDownActivatedRef.current = true
-      } else {
-        keyDownActivatedRef.current = false
-
-        requestAnimationFrame(() => {
-          // Sometimes browsers don't active the element until after the event has been processed,
-          // so we check again on the next frame
-          if (activationTarget.matches(':active')) {
+  useKeyListener({
+    onKeyDown: useCallback(
```

Comment:
Also there's no value in `useCallback`ing stuff passed to `useKeyListener`, it just stores it in refs anyway. You're just doing needless bookkeeping if you do this